miio/pwzn_relay.py

```diff
@@ -0,0 +1,380 @@
+import logging
+from collections import defaultdict
+from typing import Dict, Any, Optional
+
+import click
+
+from .click_common import command, format_output
+from .device import Device
+from .utils import deprecated
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_PWZN_RELAY_APPLE = 'pwzn.relay.apple'
+
+AVAILABLE_PROPERTIES = {
+    MODEL_PWZN_RELAY_APPLE: [
+        'relay_status', 'on_count', 'name0', 'name1', 'name2', 'name3',
+        'name4', 'name5', 'name6', 'name7', 'name8', 'name9', 'name10',
+        'name11', 'name12', 'name13', 'name14', 'name15'
+    ],
+}
+
+
+class PwznRelayStatus:
+    """Container for status reports from the plug."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """
+        Response of a PWZN Relay Apple (pwzn.relay.apple)
+        { 'relay_status': 9, 'on_count': 2, 'name0': 'channel1', 'name1': '',
+        'name2': '', 'name3': '', 'name4': '', 'name5': '', 'name6': '',
+        'name7': '', 'name8': '', 'name9': '', 'name10': '', 'name11': '',
+        'name12': '', 'name13': '', 'name14': '', 'name15': '' }
+        """
+        self.data = data
+
+    def relay(self, num: int) -> bool:
+        """Current relay X state."""
+        if "relay_status" in self.data:
+            return True if self.data["relay_status"] & (1 << num) else False
+
+    @property
+    def relay0(self) -> bool:
```

Comment:
I think it would be better to let the users of the library to call the `relay` (maybe rename to `relay_state` given it returns whether the given relay is on?).